tokio-trace-core/src/callsite.rs

```diff
@@ -52,7 +52,7 @@ pub trait Callsite: Sync {
 ///
 /// Two `Identifier`s are equal if they both refer to the same callsite.
 #[derive(Clone)]
-pub struct Identifier(&'static Callsite);
+pub struct Identifier(pub &'static Callsite);
```

Comment:
`Identifier` should not have public fields. Instead, it should probably just have a `fn from_callsite(&'static Callsite)` function.

Reply:
I believe we discussed this earlier --- in https://github.com/hawkw/tokio-trace-prototype/issues/138#issuecomment-442987155, I said that the reason `Fields` had a `&'static Callsite` rather than an `Identifier` is because it has to be constructed in a static context. When `const fn` is stable, we could provide a `fn from_callsite(&'static Callsite)` which would be sufficient, but until that point, we either need to use a `&'static Callsite` in `Fields` or have `Identifier.0` be public.

At the time, [you said](https://github.com/hawkw/tokio-trace-prototype/issues/138#issuecomment-443262170):
> In that case, I think creating an `Identifier` can be public. The details of callsites are a `tokio-trace` macro concern, so it feels ok to make `Identifier` construction public.

which I took to mean that `Identifier` _literals_ could be made public, since having a public constructor is not (currently) sufficient on stable Rust, but, looking back, I realise you never explicitly said that the field could be public.

Would it be preferable to make this field `#[doc(hidden)]` and provide a `from_callsite` constructor as well?